lib/rust/ensogl/core/src/display/navigation/navigator.rs

```diff
@@ -23,42 +23,116 @@ use crate::display::Scene;
 const DEFAULT_MAX_ZOOM: f32 = 100.0;
 /// Default minimum zoom factor (0.15x).
 const MIN_ZOOM: f32 = 0.001;
+/// Default speed for zoom events.
+const DEFAULT_ZOOM_SPEED: f32 = 0.01;
+/// Default speed for panning events.
+const DEFAULT_PAN_SPEED: f32 = 1.0;
 
 
 
+// ================
+// === Settings ===
+// ================
+
+/// Enabling/disabling Navigator and changing its settings.
+#[derive(Debug, Clone)]
+pub struct Settings {
+    is_enabled:           Cell<bool>,
+    zoom_speed:           Cell<Switch<f32>>,
+    pan_speed:            Cell<Switch<f32>>,
+    enable_wheel_panning: Cell<bool>,
+    max_zoom:             Cell<Option<f32>>,
+}
```

Comment:
During the interactive review, we've been chatting to change that in such a way that `Cell` is outside of the struct, not inside. You agreed to do it. This is the second time when I'm doing a review and not all comments from the previous review are applied, which makes this PR process extremely long. Because of this, I needed to do already 2 not necessary reviews and I will be doing one more after all comments from previous reviews will be applied. Either change it to the way we agreed on during the call or let's have another short call about why this way is better. CC @farmaazon @sylwiabr

Reply:
I believe this: 
> You agreed to do it.

 is not true. We've been chatting about it, and you said that it *might be nice* to move this `Cell` to outside of the struct. As far as I remember, you were concerned with the memory layout of this struct, but during the call, we checked the memory layout of the `Cell` (it is `transparent`) and you admitted that it is ok to use it here, but suggested refactoring it anyway. I said *I will look at it* - and I did. I don't believe such refactoring is useful.

The idea of refactoring is to use `Rc<Cell<Settings>>` instead of `Rc<Settings>` we're using now. The straightforward refactoring won't work, as `Cell` does not have a nice abstraction for changing *part* of the internal value. [`Cell::set`](https://doc.rust-lang.org/stable/std/cell/struct.Cell.html#method.set) cannot be used to change only e.g. `max_zoom` field. There are two alternatives:
1. Use `RefCell<Settings>` instead. This will allow changing parts of the struct but will bring unnecessary runtime checks.
2. Revert changes in API and move setters and getters of the `Settings` struct back to the `Navigator`. This will allow to use `Rc<Cell<Settings>>` and will keep the user-facing API simple. .The implementation of these setters will still fight with `Cell` surrounding the struct though.

So I can't see a clear solution here and I can't see any pros of the suggested refactoring.

Reply:
I would stay with Cells. Before every field was under `Rc` and nobody noticed.

Reply:
> but suggested refactoring it anyway. I said I will look at it - and I did. 

@vitvakatu if you did look and it, you should post info about it and why you want to keep it this way. In case no info is posted, I assume that the last agreement from the call is valid (refactoring it anyway).

You are right, refactoring it would require creating a bunch of setters and getters, which increases the boilerplate. In such a case we can stay with the proposed solution. However, please, always get back with thoughts/design decisions after reviews.